include/boost/math/special_functions/detail/polygamma.hpp

```diff
@@ -148,7 +148,7 @@ namespace boost { namespace math { namespace detail{
     if(iter > (int)policies::get_max_series_iterations<Policy>())
        return policies::raise_evaluation_error<T>(function, ("Exceeded maximum series evaluations evaluating at n = " + std::to_string(n) + " and x = %1%").c_str(), x, pol);
 
-    const int minus_m_minus_one = -m - 1;
+    const int minus_m_minus_one = ~m;
```

Comment:
I need someone to convince me that this doesn't rely on undefined or implementation defined behaviour (2's complement).